### PR TITLE
Stop localizing '<Unknown>' and '<In Memory Module>' strings

### PR DESCRIPTION
**Pull request by agocke**
GetFullyQualifiedName returns special strings for modules without
paths. Those names are currently localized. To improve predictability,
this change removes localization for these strings.

### CODE REVIEW

File: src/coreclr/vm/commodule.cpp [hunk 1]
```diff
@@ -862,15 +862,15 @@ void QCALLTYPE COMModule::GetFullyQualifiedName(QCall::ModuleHandle pModule, QCa
         if (*fileName != 0) {
                 retString.Set(fileName);
         } else {
-            hr = UtilLoadStringRC(IDS_EE_NAME_UNKNOWN, wszBuffer, sizeof( wszBuffer ) / sizeof( WCHAR ), true );
+            hr = UtilLoadStringRC(W("<Unknown>"), wszBuffer, sizeof( wszBuffer ) / sizeof( WCHAR ), true );
```
Comment (jkotas) [inline src/coreclr/vm/commodule.cpp:865]: This should be just `retString.Set(W("<Unknown>"))`. It should not be calling `UtilLoadStringRC` at all.

Comment (jkotas) [reply src/coreclr/vm/commodule.cpp:865]: Also, delete `wszBuffer` local in this method.

Comment (agocke) [reply src/coreclr/vm/commodule.cpp:865]: duh, thanks

**Comment by danmosemsft** [comment]
I didn't realize we localized any coreclr strings..

**Comment by agocke** [comment]
@danmosemsft Can't speak to whether or not they actually get localized!

**Comment by jkotas** [comment]
> I didn't realize we localized any coreclr strings..

It is same as localization for libraries: We have infrastructure to enable localization that we have inherited from .NET Framework, but do not actually take advantage of it (yet?).

**Review by vitek-karas — APPROVED** [review]
Do we have a test which validates that the property returns `<Unknown>` and `<In Memory Module>`. I know it won't test this specific change as that should have already worked, but since we're looking into it, we should make sure we have test coverage. (Also those tests should one day run in single-file mode and hopefully catch the problem this change is fixing)

**Comment by vitek-karas** [comment]
This should fix https://github.com/dotnet/runtime/issues/40103 - but there are issues with tests. See https://github.com/dotnet/runtime/issues/40103#issuecomment-754668138 for the details. Ideally we should solve those as well - maybe as a followup change.

**Comment by agocke** [comment]
I'm going to first see if I can get this test to fail in the lab, then re-apply the fix.

**Comment by vitek-karas** [comment]
Given that we want to take this for servicing - maybe we should just merge the fix without test coverage (as that is tricky to get right) - start the servicing process - and in the meantime work on the tests.

**Comment by agocke** [comment]
Agreed -- I've actually found a number of test problems here that's going to take a while to sort out. I'll bring the change back and open a separate PR for testing.

**Comment by agocke** [comment]
Got the test to fail locally, and then confirmed this fixes it. I think this is ready for merge and backport. @jkotas Anything else?

**Review by jkotas — APPROVED** [review]
LGTM

**Pull request merged**